### DISCUSSION TITLE
Implement collision and contact

### DISCUSSION THREAD

**Discussion by Simone2021space** [sympy/sympy#22309] [Q&A]
Hi team,
is there a sympy tool for implementing collisions?
Like a function that takes in input two bodies and returns the contact force.

I know the force is related to bodies reciprocal velocities but I don't know the correct formula, do you have any reference for it? 

Further, is it possible to model the surface of a rigid-body, how?

**Comment by moorepants** [comment]
No forces or collisions are defined in sympy.physics.mechanics. In general, the user is expected to define the math of any forces they may want to add to the model. Most advanced dynamics textbooks detail collision.

If you want prebaked bodies with collision surfaces you may want to look into physics engines.

**Reply by Simone2021space** [reply]
Many thanks.
I will look at Kane's book. Any suggestion about other textbooks would be highly appreciated.

Are you aware of any physics engines compatible with sympy? I am wondering if it is possible to write the symbolic EoMs in sympy and than feed them to the engine for running the simulation.

**Reply by moorepants** [reply]
> I will look at Kane's book

He has a small section on collision, but may not be enough for detailed needs.

> Any suggestion about other textbooks would be highly appreciated.

Looking at papers might give the best options. I don't know of a book that really explains it well for modern computational dynamics.

> Are you aware of any physics engines compatible with sympy?

No, most physics engines I know of expect you to construct the problem with their system and it generates EoMs. If you have the EoMs already, then the only part left is the simulation which is a bit minor, in some sense.
